src/test/regress/expected/coordinator_shouldhaveshards.out

```diff
@@ -895,6 +895,145 @@ NOTICE:  executing the command locally: SELECT count(*) AS count FROM (SELECT in
+ count
+---------------------------------------------------------------------
+     0
+(1 row)
+
+  WITH cte_1 as (SELECT * FROM test LIMIT 5) SELECT count(*) FROM test;
+NOTICE:  executing the command locally: SELECT count(*) AS count FROM coordinator_shouldhaveshards.test_1503000 test WHERE true
+NOTICE:  executing the command locally: SELECT count(*) AS count FROM coordinator_shouldhaveshards.test_1503003 test WHERE true
+ count
+---------------------------------------------------------------------
+     0
+(1 row)
+
+  SELECT y FROM test WHERE x = 1;
+NOTICE:  executing the command locally: SELECT y FROM coordinator_shouldhaveshards.test_1503000 test WHERE (x OPERATOR(pg_catalog.=) 1)
+ y
+---------------------------------------------------------------------
+(0 rows)
+
+  WITH cte_1 AS (SELECT y FROM test WHERE x = 1 LIMIT 5) SELECT count(*) FROM test;
+NOTICE:  executing the command locally: SELECT count(*) AS count FROM coordinator_shouldhaveshards.test_1503000 test WHERE true
+NOTICE:  executing the command locally: SELECT count(*) AS count FROM coordinator_shouldhaveshards.test_1503003 test WHERE true
+ count
+---------------------------------------------------------------------
+     0
+(1 row)
+
+  SELECT coordinated_transaction_uses_2PC();
+ coordinated_transaction_uses_2pc
+---------------------------------------------------------------------
+ f
+(1 row)
+
+COMMIT;
+-- a local SELECT followed by a remote Modify
+-- triggers 2PC
+BEGIN;
+  SELECT y FROM test WHERE x = 1;
+NOTICE:  executing the command locally: SELECT y FROM coordinator_shouldhaveshards.test_1503000 test WHERE (x OPERATOR(pg_catalog.=) 1)
+ y
+---------------------------------------------------------------------
+(0 rows)
+
+  UPDATE test SET y = y +1;
+NOTICE:  executing the command locally: UPDATE coordinator_shouldhaveshards.test_1503000 test SET y = (y OPERATOR(pg_catalog.+) 1)
+NOTICE:  executing the command locally: UPDATE coordinator_shouldhaveshards.test_1503003 test SET y = (y OPERATOR(pg_catalog.+) 1)
+  SELECT coordinated_transaction_uses_2PC();
+ coordinated_transaction_uses_2pc
+---------------------------------------------------------------------
+ t
+(1 row)
+
+COMMIT;
+-- a local modify followed by a remote SELECT
+-- triggers 2PC
+BEGIN;
+  INSERT INTO test VALUES (1,1);
+NOTICE:  executing the command locally: INSERT INTO coordinator_shouldhaveshards.test_1503000 (x, y) VALUES (1, 1)
+  SELECT count(*) FROM test;
+NOTICE:  executing the command locally: SELECT count(*) AS count FROM coordinator_shouldhaveshards.test_1503000 test WHERE true
+NOTICE:  executing the command locally: SELECT count(*) AS count FROM coordinator_shouldhaveshards.test_1503003 test WHERE true
+ count
+---------------------------------------------------------------------
+     1
+(1 row)
+
+  SELECT coordinated_transaction_uses_2PC();
+ coordinated_transaction_uses_2pc
+---------------------------------------------------------------------
+ t
```

Comment:
should we merge https://github.com/citusdata/citus/pull/4587 and rebase?

Reply:
I think this test wouldn't change, but we would prevent actually  doing 2PC for remote connections.

I'd rather merge this and test that on top of this as well, if that works?

Reply:
Why do we have 2PC here then?

Reply:
I wonder if we should rename the variable to ShouldUse2PC

Reply:
Or, if I follow your other comment, this should be false. I think that makes sense